src/server/frontend_wayland/wayland_connector.h

```diff
@@ -131,7 +136,7 @@ class WaylandConnector : public Connector
     std::unique_ptr<DataDeviceManager> data_device_manager_global;
     std::shared_ptr<Executor> const executor;
     std::shared_ptr<graphics::WaylandAllocator> const allocator;
-    std::weak_ptr<frontend::Shell> const weak_shell;
+    std::weak_ptr<shell::Shell> const weak_shell;
```

Comment:
This is arguably pre-existing, but why would the frontend have a weak pointer to the shell?

The shell should have no dependency on the frontend(s), so having a strong pointer seems safe and correct.